packages/base-controller/src/RestrictedControllerMessenger.ts

```diff
@@ -95,7 +111,6 @@ export class RestrictedControllerMessenger<
   registerActionHandler<
     ActionType extends Action['type'] & NamespacedName<Namespace>,
   >(action: ActionType, handler: ActionHandler<Action, ActionType>) {
-    /* istanbul ignore if */ // Branch unreachable with valid types
     if (!this.#isInCurrentNamespace(action)) {
```

Comment:
Shouldn't it be possible to register handlers for allowed actions?

```suggestion
    if (!this.#isAllowedAction(action)) {
```